nanoFramework.System.Net.Http/Http/HttpClient.cs

```diff
@@ -0,0 +1,561 @@
+        /// <exception cref="ArgumentException">Value is null or it not an absolute Uniform Resource Identifier (URI).</exception>
+        /// <exception cref="InvalidOperationException">An operation has already been started on the current instance.</exception>
+        /// <exception cref="ObjectDisposedException">The current instance has been disposed.</exception>
+        public TimeSpan Timeout
+        {
+            get => _timeout;
+
+            set
+            {
+                if (value != Threading.Timeout.InfiniteTimeSpan && (value <= TimeSpan.Zero || value.TotalMilliseconds > int.MaxValue))
+                {
+                    throw new ArgumentOutOfRangeException();
+                }
+
+                CheckDisposedOrStarted();
+
+                _timeout = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the root CA certificate used to authenticate with https servers.
+        /// This certificate is used only for https connections; http connections do not require this.
+        /// </summary>
+        /// <remarks>
+        /// This property is an extension from the full .NET required by nanoFramework.
+        /// </remarks>
+        public X509Certificate HttpsAuthentCert { get; set; }
+
+        /// <summary>
+        /// Gets or sets the TLS/SSL protocol used by the <see cref="HttpClient"/> class.
+        /// </summary>
+        /// <value>
+        /// One of the values defined in the <see cref="Security.SslProtocols"/> enumeration. Default value is <see cref="SslProtocols.Tls12"/>.
+        /// </value>
+        /// <remarks>
+        /// This property is an extension from the full .NET required by nanoFramework.
+        /// </remarks>
+        public SslProtocols SslProtocols { get; set; } = SslProtocols.Tls12;
+
+        #region Constructors
+
+        public HttpClient() : base(new HttpClientHandler(), true)
+        {
+            _timeout = Threading.Timeout.InfiniteTimeSpan;
+            //_maxResponseContentBufferSize = HttpContent.MaxBufferSize;
+        }
+
+        #endregion Constructors
+
+        #region REST Overloads
+
+        /// <summary>
+        /// Send a DELETE request to the specified Uri as a synchronous operation.
+        /// </summary>
+        /// <param name="requestUri">The <see cref="Uri"/> the request is sent to.</param>
+        /// <returns>The <see cref="HttpResponseMessage"/> object resulting from the HTTP request.</returns>
+        /// <exception cref="ArgumentNullException">The request is <see langword="null"/>.</exception>
+        /// <exception cref="InvalidOperationException">The request message was already sent by the <see cref="HttpClient"/> instance.</exception>
+        /// <exception cref="HttpRequestException">The request failed due to an underlying issue such as network connectivity, DNS failure, or server certificate validation.</exception>
+        /// <remarks>
+        /// <para>
+        /// This operation will block.
```

Comment:
for all those blocking operations, is there a way to have them canceled? Is the timeout going to play this role (sorry, reviewing without going yet thru everything)

Reply:
Not possible. We have no control at that level on the calls to native sockets. Only thing is the socket timeout.
That's why I haven't put there the CancellationToken param.

Reply:
ok, fair enough. So if the socket timeout is setup properly, then the method will be stopped? Correct?.